strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/PypiWheelArtifactCoordinates.java

```diff
@@ -0,0 +1,239 @@
+package org.carlspring.strongbox.artifact.coordinates;
+
+import javax.persistence.Entity;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.util.Map;
+
+import org.carlspring.strongbox.util.PypiWheelArtifactCoordinatesUtils;
+import org.semver.Version;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Represents {@link ArtifactCoordinates} for PyPi repository
+ * 
+ * Proper path for this coordinates is in the format of: 
+ * {distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl.
+ * Example: distribution-1.0-1-py27-none-any.whl
+ * 
+ * @author alecg956
+ */
+@Entity
+@SuppressWarnings("serial")
+@XmlRootElement(name = "PypiWheelArtifactCoordinates")
+@XmlAccessorType(XmlAccessType.NONE)
+@ArtifactCoordinatesLayout(name = PypiWheelArtifactCoordinates.LAYOUT_NAME, alias = PypiWheelArtifactCoordinates.LAYOUT_ALIAS)
+public class PypiWheelArtifactCoordinates
+    extends AbstractArtifactCoordinates<PypiWheelArtifactCoordinates, Version>
+{
+
+    public static final String LAYOUT_NAME = "PyPi";
+    public static final String LAYOUT_ALIAS = "pypi";
+
+    public static final String DISTRIBUTION = "distribution";
+    public static final String VERSION = "version";
+    public static final String BUILD_TAG = "build_tag";
+    public static final String LANG_IMPL_VERSION_TAG = "languageImplementationVersion_tag";
+    private static final String ABI_TAG = "abi_tag";
+    private static final String PLATFORM_TAG = "platform_tag";
+
+    /**
+     * This method takes in all artifact coordinates of a PyPi Wheel filename, with build being 
+     * the empty string if it is not included in the filename
+     * 
+     * @param distribution: Uniquely identifying artifact coordinate (required)
+     * @param version: Packages current version (required)
+     * @param build: Build_tag parameter (optional)
+     * @param languageImplementationVersion: Language and Implementation version argument (required)
+     * @param abi: ABI tag parameter (required)
+     * @param platform: Platform tag parameter (required)
+     * @return bar Returns bar
+     */
+    public PypiWheelArtifactCoordinates(String distribution,
+                                        String version,
+                                        String build,
+                                        String languageImplementationVersion,
+                                        String abi,
+                                        String platform)
+    {
+        // if any of the required arguments are empty, throw an error
+        if (distribution == "" || version == "" || languageImplementationVersion == "" || abi == "" || platform == "")
```

Comment:
In Java, comparing objects should be done using `equals()`. Checking using `==` checks that two operands point to the same object. Have a look at these articles:
* https://stackoverflow.com/a/13387787/774183
* https://www.leepoint.net/data/expressions/22compareobjects.html
* https://www.geeksforgeeks.org/equals-hashcode-methods-java/

Reply:
Resolved.